### PR TITLE
OSDOCS#7991: Operator SDK 1.31 migration docs

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Version(s):  4.14

Issue: [OSDOCS-7991](https://issues.redhat.com//browse/OSDOCS-7991)

Link to docs preview:
* [Updating Go-based Operator projects](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/golang/osdk-golang-updating-projects)
* [Updating Java-based Operator projects](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/java/osdk-java-updating-projects)
* [Updating Hybrid-Helm-based Operator projects](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/helm/osdk-hybrid-helm-updating-projects)
* [Updating Helm-based Operator projects](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/helm/osdk-helm-updating-projects)
* [Updating Ansible-based Operator projects](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/ansible/osdk-ansible-updating-projects)

Misc. docs where `osdk_ver` is updated. These docs are updated by the the change to the attribute file. There are no corresponding files to these docs in the pull request.
* [Installing the Operator SDK CLI](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-installing-cli)
* [Running the bundle validate command](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-bundle-validate#osdk-bundle-validate-run_osdk-bundle-validate)
* [Enabling Operator conditions](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-generating-csvs#osdk-operatorconditions_osdk-generating-csvs)
** NOTE: This lists OSDK 1.31 compatibility with `operator-lib` 0.11.0`
* Scorecard
** [Scorecard configuration](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-scorecard#osdk-scorecard-config_osdk-scorecard) (Example config files)
** [Scorecard output](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-scorecard#osdk-scorecard-output_osdk-scorecard)
** [Scorecard parallel testing](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-scorecard#osdk-scorecard-parallel_osdk-scorecard) (first step in the procedure)
* [About Operator SDK](https://65611--docspreview.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-about) (NOTE admonition)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/osdk-updating-128-to-131.adoc [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+.Example Dockerfile
+[source,docker]
+----
+FROM quay.io/operator-framework/ansible-operator:v{osdk_ver} <1>
+----
+<1> Update the Ansible Operator version from `{osdk_ver_n1}` to `{osdk_ver}`
++
+[WARNING]
+----
+The update to Ansible 2.15.0 in version 1.30.0 of the Ansible Operator removed the following pre-installed Python modules:
```
Comment (michaelryanpeter) [inline modules/osdk-updating-128-to-131.adoc:118]: Does we also have to manually update dependencies in the `[Pipfile](https://github.com/operator-framework/operator-sdk/pull/6483/files#diff-85c2285c39846cfb1ab26c4373213d7c91c4705e10d590b41c4d9503127fe812)`? Or are these version bumps handled by updating the SDK and/or Ansible Operator version?

File: modules/osdk-updating-128-to-131.adoc [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+ifdef::golang,hybrid,java[]
+* Edit your Operator project's makefile to update the Operator SDK version to {osdk_ver} as shown in the following example:
++
+.Example makefile
+[source,make]
+----
+# Set the Operator SDK version to use. By default, what is installed on the system is used.
+# This is useful for CI or a project to utilize a specific version of the operator-sdk toolkit.
+OPERATOR_SDK_VERSION ?= v{osdk_ver} <1>
```
Comment (michaelryanpeter) [inline modules/osdk-updating-128-to-131.adoc:51]: In the upstream SDK project migration docs, the migrations for the 1.31.0 release all use 1.30.0 for the version. This applies to `OPERATOR_SDK_VERSION` in the makefiles, and for the versions of the Helm and Ansible Operators. 

Should this doc also use 1.30.0? Or is it safe to assume that 1.31.0 is correct?

Comment (everettraven) [reply modules/osdk-updating-128-to-131.adoc:51]: 1.31.0 should be correct. I'm assuming there was a step missed somewhere that resulted in it still being v1.30.0, but it should be 1.31.0

File: modules/osdk-updating-128-to-131.adoc (hunk 1 above)
Comment (everettraven) [reply modules/osdk-updating-128-to-131.adoc:118]: The version bumps should be handled by updating the ansible operator base image version

File: modules/osdk-updating-128-to-131.adoc [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+- IMAGE_TARGET_LIST = operator-sdk helm-operator ansible-operator ansible-operator-2.11-preview scorecard-test scorecard-test-kuttl
++ IMAGE_TARGET_LIST = operator-sdk helm-operator ansible-operator scorecard-test scorecard-test-kuttl <1>
+  image-build: $(foreach i,$(IMAGE_TARGET_LIST),image/$(i)) ## Build all images.
+
+  # Convenience wrapper for building dependency base images.
+  .PHONY: image-build-base
+- IMAGE_BASE_TARGET_LIST = ansible-operator ansible-operator-2.11-preview
++ IMAGE_BASE_TARGET_LIST = ansible-operator <2>
+  image-build-base: $(foreach i,$(IMAGE_BASE_TARGET_LIST),image-base/$(i)) ## Build all images.
+  # ...
```
Comment (everettraven) [inline modules/osdk-updating-128-to-131.adoc:157]: This looks like an operator-sdk  makefile thing, not one scaffolded with ansible operators. I'd replace this with something to say that usage of the ansible-operator-2.11-preview base image should be replaced with the latest ansible operator base image if there isn't already a section for that (leaving these comments as I go).

**Comment by michaelryanpeter** [comment]
@everettraven Thanks for the feedback! I have implemented the updates, PTAL.

**Review by everettraven — APPROVED** [review]
LGTM!

**Comment by michaelryanpeter** [comment]
@emmajiafan or @KeenonLee, PTAL.

File: modules/osdk-updating-128-to-131.adoc [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+   - name: cloud.common
+----
+<1> Remove the `community.kubernetes` collection
+<2> Update the `operator_sdk.util` collection to version `0.5.0`.
+
+. Remove all instances of the `lint` field from your `molecule/kind/molecule.yml` and `molecule/default/molecule.yml` files, as shown in the following example:
++
+[source,diff]
+----
+  ---
```
Comment (skrthomas) [inline modules/osdk-updating-128-to-131.adoc:162]: Is this `---` intentional? Not asking for any other reason than I don't see a corresponding closing `---` (and maybe there isn't one).

Comment (skrthomas) [reply modules/osdk-updating-128-to-131.adoc:162]: I was also wondering about the `.` on line 169. Just interesting to see a period but I'm amazed everyday but the things idk so... 🙃

File: modules/osdk-updating-128-to-131.adoc [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+-    version: "0.4.0"
++    version: "0.5.0" <2>
+   - name: kubernetes.core
+     version: "2.4.0"
+   - name: cloud.common
+----
+<1> Remove the `community.kubernetes` collection
+<2> Update the `operator_sdk.util` collection to version `0.5.0`.
+
+. Remove all instances of the `lint` field from your `molecule/kind/molecule.yml` and `molecule/default/molecule.yml` files, as shown in the following example:
```
Comment (skrthomas) [inline modules/osdk-updating-128-to-131.adoc:158]: Despite the fact that this step looks off in the files, it renders OK. Not sure why its gray here.

**Review by skrthomas — COMMENTED** [review]
Only a couple of questions. No concrete things here that I saw. Looks great, nice work :)

File: modules/osdk-updating-128-to-131.adoc (hunk 4 above)
Comment (michaelryanpeter) [reply modules/osdk-updating-128-to-131.adoc:162]: The `---` is intentional, though I think it is an Ansible thing. You can see it in the wild in [here](https://github.com/operator-framework/ansible-operator-plugins/blob/main/testdata/memcached-molecule-operator/requirements.yml). And I think the `.` is related to a JSON query in linter. Good attention to detail, though!

**Pull request merged**